src/ProjectNeon/Assets/Features/Battle/Cards/PlayedCard.cs

```diff
@@ -20,4 +20,16 @@ public PlayedCard(Member performer, Target[] targets, Card card)
         _targets = targets;
         _card = card;
     }
+
+    public void perform()
+    {
+        int index = 0;
+        _card.Actions.ForEach(
```

Comment:
It's very unusual to mix indexing with ForEach. 

Why not just do a nested ForEach?

```
_card.Actions
    .ForEach(action => _targets
        .ForEach(target => action.Apply(_performer, target));
```

Reply:
@SilasReinagel Take a look to the puzzle in `PlayedCard`: I have to keep the index because target and card effects relation is maintained by the index of the respective collections (1st effect applies to 1st target, 2nd effect to 2nd target, etc).

If I use the suggested code I would apply all effects / actions to all targets, something that we doesn't want.

Reply:
@paulodamaso Ah yes. So, in that case you should just use a traditional indexer, and not use `ForEach`. 

It's always a code smell to see two different type of iteration in the same method.